adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/config/KeycloakWebSecurityConfigurerAdapter.java

```diff
@@ -117,6 +123,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 .and()
                 .addFilterBefore(keycloakPreAuthActionsFilter(), LogoutFilter.class)
                 .addFilterBefore(keycloakAuthenticationProcessingFilter(), BasicAuthenticationFilter.class)
+                .addFilterBefore(keycloakAuthenticatedActionsFilter(), BasicAuthenticationFilter.class)
```

Comment:
I just made a test, even without this line, the bean is registered and the filter is working, any idea why ? And so, do we need to explicitly add the filter ?

Reply:
Arg ;-) 
I do not see this filter registered or do not find a call to it's Constructor in the source code.

Reply:
Maybe I'm doing something weird, can you test on your side ?

Reply:
I think this answers our question : https://stackoverflow.com/a/39314867/3860046 
So 2 options : 
* Not annoting `@bean` the filter
* Disable the registration as filter : 
```
@Bean
public FilterRegistrationBean registration(MyFilter filter) {
    FilterRegistrationBean registration = new FilterRegistrationBean(filter);
    registration.setEnabled(false);
    return registration;
}
```

Reply:
I think option 2 is the best but we should just document it and not add the FilterRegistrationBean code. (We already document it for `keycloakAuthenticationProcessingFilterRegistrationBean` and `keycloakPreAuthActionsFilterRegistrationBean` )

Reply:
I wonder if we should not add a subclass of `KeycloakWebSecurityConfigurerAdapter` , like `SpringBootKeycloakWebSecurityConfigurerAdapter` that would just contain : 
``` 
    @Bean
    public FilterRegistrationBean keycloakAuthenticationProcessingFilterRegistrationBean(
            KeycloakAuthenticationProcessingFilter filter) {
        FilterRegistrationBean registrationBean = new FilterRegistrationBean(filter);
        registrationBean.setEnabled(false);
        return registrationBean;
    }

    @Bean
    public FilterRegistrationBean keycloakPreAuthActionsFilterRegistrationBean(
            KeycloakPreAuthActionsFilter filter) {
        FilterRegistrationBean registrationBean = new FilterRegistrationBean(filter);
        registrationBean.setEnabled(false);
        return registrationBean;
    }
   @Bean
    public FilterRegistrationBean keycloakAuthenticatedActionsFilterRegistrationBean(
            KeycloakAuthenticatedActionsFilter filter) {
        FilterRegistrationBean registrationBean = new FilterRegistrationBean(filter);
        registrationBean.setEnabled(false);
        return registrationBean;
    }

```

Reply:
I found another issue : if we don't provide any bearer token, the response is still correct but under the hood we are getting a class cast exception because the `AnonymousAuthenticationFilter` returns a `AnonymousAuthenticationToken` . We can avoid that by adding in the `configure` method : ` http.anonymous().disable()` but not sure it's a good idea. Wdyt ?

Reply:
@trecloux Did you had time to view my comments ?

Reply:
@sebastienblanc, after our discussions and tests, I created issue KEYCLOAK-5808 to cover the spring boot configuration class

Reply:
@sebastienblanc I'm also facing the issue with the AnonymousAuthenticationToken as I want to allow certain endpoints to be reachable anonymously and limiting ip addresses. Any hint how to solve such a scenario using the spring security keycloak adapter?

```java
    @Override
    public void configure(HttpSecurity http) throws Exception
    {
        super.configure(http);
        http
                .cors()
                .and()
                .csrf() //Only for dev purposes
                .disable()
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                .sessionAuthenticationStrategy(sessionAuthenticationStrategy())
                .and()
                .authorizeRequests()
                .antMatchers("/api/user*").hasRole("USER")
                .antMatchers("/api/admin*").hasRole("ADMIN")
                .antMatchers("/api/nouser").hasIpAddress("127.0.0.1")
                .anyRequest().permitAll();
    }
```